lib/java/com/google/android/material/bottomsheet/BottomSheetDialogFragment.java

```diff
@@ -31,4 +39,94 @@
   public Dialog onCreateDialog(Bundle savedInstanceState) {
     return new BottomSheetDialog(getContext(), getTheme());
   }
+
+  @Override
+  public void dismiss() {
+    Dialog dialog = getDialog();
+    if (dialog != null) {
+      View bottomSheet = dialog.findViewById(R.id.design_bottom_sheet);
+      if (bottomSheet != null) {
+        BottomSheetBehavior<View> behavior = BottomSheetBehavior.from(bottomSheet);
+        if (behavior.dismissWithAnimation) {
```

Comment:
@cketcham Commenting here for visibility.
I was testing some stuff locally, and it looks like if the behavior has
`isHidable = false` and we use the `BottomSheetBehavior.STATE_HIDDEN`, an exception is thrown here:

https://github.com/material-components/material-components-android/blob/master/lib/java/com/google/android/material/bottomsheet/BottomSheetBehavior.java#L1127

I think we should add one my check in this `if` statement:

```
if (behavior.isHidable() && behavior.dismissWithAnimation) { 
  dismissWithAnimation(behavior)
} else {
  super.dismiss()
}